gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/testing/LocalPubsubHelper.java

```diff
@@ -0,0 +1,83 @@
+package com.google.gcloud.pubsub.testing;
+
+import com.google.gcloud.pubsub.spi.testing.LocalPublisherImpl;
+import com.google.pubsub.v1.PublisherGrpc;
+
+import io.grpc.ManagedChannel;
+import io.grpc.Server;
+import io.grpc.netty.NegotiationType;
+import io.grpc.netty.NettyChannelBuilder;
+import io.grpc.netty.NettyServerBuilder;
+import io.netty.channel.local.LocalAddress;
+import io.netty.channel.local.LocalChannel;
+import io.netty.channel.local.LocalServerChannel;
+
+import java.io.IOException;
+import java.net.SocketAddress;
+
+/**
+ * A class that runs an in-memory Publisher instance for use in tests.
+ */
+public class LocalPubsubHelper {
+  private static final SocketAddress address = new LocalAddress("in-process-1");
+  private static Server server;
```

Comment:
what "pattern of server" are you referring to?

Other methods, such as `reset` do not return `this` (and for a good reason).
I think you should either make `start` a factory method accepting addressString and returning a new instance of `LocalPubsubHelper` or make it `void`, anything else feels inconsistent to me.